sdk/graphql-transport/src/queries/objects.graphql

```diff
@@ -0,0 +1,200 @@
+# # Copyright (c) Mysten Labs, Inc.
+# # SPDX-License-Identifier: Apache-2.0
+
+query getOwnedObjects(
+	$owner: SuiAddress!
+	$limit: Int
+	$cursor: String
+	$showBcs: Boolean = false
+	$showContent: Boolean = false
+	$showDisplay: Boolean = false
+	$showType: Boolean = false
+	$showOwner: Boolean = false
+	$showPreviousTransaction: Boolean = false
+	$showStorageRebate: Boolean = false
+	$filter: ObjectFilter
+) {
+	address(address: $owner) {
+		objects(first: $limit, after: $cursor, filter: $filter) {
+			pageInfo {
+				hasNextPage
+				endCursor
+			}
+			nodes {
+				...RPC_MOVE_OBJECT_FIELDS
+			}
+		}
+	}
+}
+
+query getObject(
+	$id: SuiAddress!
+	$showBcs: Boolean = false
+	$showOwner: Boolean = false
+	$showPreviousTransaction: Boolean = false
+	$showContent: Boolean = false
+	$showDisplay: Boolean = false
+	$showType: Boolean = false
+	$showStorageRebate: Boolean = false
+) {
+	object(address: $id) {
+		...RPC_OBJECT_FIELDS
+	}
+}
+
+query tryGetPastObject(
+	$id: SuiAddress!
+	$version: Int
+	$showBcs: Boolean = false
+	$showOwner: Boolean = false
+	$showPreviousTransaction: Boolean = false
+	$showContent: Boolean = false
+	$showDisplay: Boolean = false
+	$showType: Boolean = false
+	$showStorageRebate: Boolean = false
+) {
+	current: object(address: $id) {
+		address
+		version
+	}
+	object(address: $id, version: $version) {
+		...RPC_OBJECT_FIELDS
+	}
+}
+
+query multiGetObjects(
+	$ids: [SuiAddress!]!
+	$limit: Int
+	$cursor: String
+	$showBcs: Boolean = false
+	$showContent: Boolean = false
+	$showDisplay: Boolean = false
+	$showType: Boolean = false
+	$showOwner: Boolean = false
+	$showPreviousTransaction: Boolean = false
+	$showStorageRebate: Boolean = false
+) {
+	objects(first: $limit, after: $cursor, filter: { objectIds: $ids }) {
+		pageInfo {
+			hasNextPage
+			endCursor
+		}
+		nodes {
+			...RPC_OBJECT_FIELDS
+		}
+	}
+}
+
+fragment RPC_OBJECT_FIELDS on Object {
+	objectId: address
+	bcs @include(if: $showBcs)
+	version
+	asMoveObject @include(if: $showType) {
+		contents {
+			type {
+				repr
+			}
+		}
+	}
+	asMoveObject @include(if: $showContent) {
+		hasPublicTransfer
+		contents {
+			data
+			type {
+				repr
+				layout
+				signature
+			}
+		}
+	}
+	asMoveObject @include(if: $showBcs) {
+		hasPublicTransfer
+		contents {
+			bcs
+			type {
+				repr
+			}
+		}
+	}
+	owner @include(if: $showOwner) {
+		...RPC_OBJECT_OWNER_FIELDS
+	}
+	previousTransactionBlock @include(if: $showPreviousTransaction) {
+		digest
+	}
+
+	storageRebate @include(if: $showStorageRebate)
+	digest
+	version
+	display @include(if: $showDisplay) {
+		key
+		value
+		error
+	}
+}
+
+fragment RPC_MOVE_OBJECT_FIELDS on MoveObject {
+	objectId: address
+	bcs @include(if: $showBcs)
+	version @include(if: $showType)
```

Comment:
we have a conditional select here, but we also unconditionally select `version` below

Reply:
good catch